### PR TITLE
Cart page

### PR DESCRIPTION
**Pull request by yashodaAgarwal**
Have added cart functionality in the ecom. Please review the code

### CODE REVIEW

File: src/pages/cart-page/CartPage.jsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    setOriginalTotal(
+      cart.reduce((acc, curr) => acc + Number(curr.originalPrice * curr.qty), 0)
+    );
+    setTotal(
+      cart.reduce((acc, curr) => acc + Number(curr.price * curr.qty), 0)
+    );
+    setDiscount(
+      cart.reduce(
+        (acc, curr) =>
+          acc + Number((curr.originalPrice - curr.price) * curr.qty),
```
Comment (Debarshi95) [inline src/pages/cart-page/CartPage.jsx:27]: Repeating the cart the reduce function. You can extract the code to a custom function and use it here. The way the code will become more readable

File: src/pages/cart-page/CartPage.jsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
+import { useState } from "react";
+import { useEffect } from "react";
+import { Navigation } from "../../components";
+import { useProduct } from "../../context";
+import "./cart-page.css";
```
Comment (Debarshi95) [inline src/pages/cart-page/CartPage.jsx:5]: Instead of  .. /../, use absolute imports

File: src/pages/cart-page/CartPage.jsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  }, [cart]);
+
+  return (
+    <div>
+      <Navigation />
+      <div className="mycart-container">
+        <div className="mycart-heading ">Shopping Cart ({cart.length})</div>
+        {cart.map((item) => {
+          return (
+            <div className="flex">
```
Comment (Debarshi95) [inline src/pages/cart-page/CartPage.jsx]: Key is missing

**Review by Debarshi95 — APPROVED** [review]
Good work. All okay for now. Few changes suggested, may be work upon them on future releases.

**Review by saurabhchirde — APPROVED** [review]
Couldn't find any issue other than those suggestions given by @Debarshi95 
Looks good to merge.

**Pull request merged**